src/mbgl/text/collision_feature.cpp

```diff
@@ -35,7 +36,23 @@ CollisionFeature::CollisionFeature(const GeometryCoordinates& line,
         GeometryCoordinate anchorPoint = convertPoint<int16_t>(anchor.point);
         bboxifyLabel(line, anchorPoint, anchor.segment, length, height, overscaling);
     } else {
-        boxes.emplace_back(anchor.point, Point<float>{ 0, 0 }, x1, y1, x2, y2);
+        if (rotate) {
+            const float rotateRadians = rotate * M_PI / 180.0;
+
+            const Point<float> tl = util::rotate(Point<float>(x1, y1), rotateRadians);
+            const Point<float> tr = util::rotate(Point<float>(x2, y1), rotateRadians);
+            const Point<float> bl = util::rotate(Point<float>(x1, y2), rotateRadians);
+            const Point<float> br = util::rotate(Point<float>(x2, y2), rotateRadians);
+            
+            const float xMin = std::min({tl.x, tr.x, bl.x, br.x});
```

Comment:
Could you bring over the comments from GL JS here or something similar? I think the "we have to represent on-axis geometry, so take the envelope of the rotated geometry" step is fairly non-obvious to someone casually browsing the code.